app/services/CategorisationService.scala

```diff
@@ -87,15 +87,35 @@ class CategorisationService @Inject() (
         Category2Scenario
       }
     } else {
-      calculateResultWithoutNiphl(categorisationInfo, userAnswers, recordId)
+      calculateResultWithNirms(categorisationInfo, userAnswers, recordId, listOfAnswers)
     }
   }
 
-  private def calculateResultWithoutNiphl(
+  private def calculateResultWithNirms(
+    categorisationInfo: CategorisationInfo,
+    userAnswers: UserAnswers,
+    recordId: String,
+    listOfAnswers: Seq[AnsweredQuestions]
+  ): Scenario = {
+
+    val areThereQuestionsWithNoExemption = listOfAnswers.exists(x => x.answer.contains(AssessmentAnswer.NoExemption))
+
+    if (categorisationInfo.isNirmsAssessment && !areThereQuestionsWithNoExemption && listOfAnswers.nonEmpty) {
```

Comment:
the listOfAnswers.nonEmpty seems unnecessary? 
consider case where my commodity has one assessment question, which is a Nirms question

If I do not have Nirms, we get the right answer Category 2(accidentally?) - we go into "calculateWithoutNiphlsNirms" and falls into Category 2

If I do have Nirms, we get the wrong answer Category 2 - again goes into calculateWIthoutNiphlsNirms (which suggests a test case missing!)

This tweak might solve it (not fully tested it - but it looks more like the niphls one we know works :) )


```
  private def calculateResultWithNirms(
    categorisationInfo: CategorisationInfo,
    userAnswers: UserAnswers,
    recordId: String,
    listOfAnswers: Seq[AnsweredQuestions]
  ): Scenario = {

    val areThereCategory1AnsweredNo = listOfAnswers.exists(ass => ass.answer.contains(AssessmentAnswer.NoExemption) && ass.question.isCategory1)
    val areThereCategory2AnsweredNo = listOfAnswers.exists(ass => ass.answer.contains(AssessmentAnswer.NoExemption) && ass.question.isCategory2)
    val areThereCategory1Unanswerable = categorisationInfo.categoryAssessments.exists(ass => ass.isCategory1 && ass.hasNoAnswers)
    val areThereCategory2Unanswerable = categorisationInfo.categoryAssessments.exists(ass => ass.isCategory2 && ass.hasNoAnswers)

    if (categorisationInfo.isNirmsAssessment && !areThereCategory1Unanswerable) {

      if (areThereCategory1AnsweredNo) {
        Category1Scenario
      } else if (!categorisationInfo.isTraderNirmsAuthorised || areThereCategory2AnsweredNo || areThereCategory2Unanswerable) {
        Category2Scenario
      } else {
        StandardGoodsScenario
      }
    } else {
      calculateResultWithoutNiphlAndNirms(categorisationInfo, userAnswers, recordId)
    }
  }```

Reply:
Updated to the new logic.